### PR TITLE
bom: inherit project; use flatten plugin to sanitize bom

### PR DESCRIPTION
**Pull request by stevenschlansker**
This PR removes duplicate declarations from the `bom` artifact.  Instead, we inherit it from `jetty-project` and use the `maven-flatten-plugin` to sanitize the bom before publishing.

Here is a diff of the changes to the final bom:

`[steven@nevets:~/code/jetty.project/jetty-bom](flatten-bom-9.4|…)% python ~/xmldiffs.py /Users/steven/.m2/repository/org/eclipse/jetty/jetty-bom/9.4.14-SNAPSHOT/jetty-bom-9.4.14-SNAPSHOT.pom flattened-pom.xml | sed 's%{http://maven.apache.org/POM/4.0.0}%%g'`
```diff
--- /Users/steven/.m2/repository/org/eclipse/jetty/jetty-bom/9.4.14-SNAPSHOT/jetty-bom-9.4.14-SNAPSHOT.pom
+++ flattened-pom.xml
@@ -1,73 +1,7 @@
-<project {http://www.w3.org/2001/XMLSchema-instance}schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+<project {http://www.w3.org/2001/XMLSchema-instance}schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <artifactId>
   jetty-bom
   </artifactId>
-  <build>
-    <pluginManagement>
-      <plugins>
-        <plugin>
-          <artifactId>
-          maven-release-plugin
-          </artifactId>
-          <configuration>
-            <arguments>
-            -Peclipse-release
-            </arguments>
-            <goals>
-            deploy
-            </goals>
-            <mavenExecutorId>
-            forked-path
-            </mavenExecutorId>
-            <preparationGoals>
-            clean install
-            </preparationGoals>
-            <useReleaseProfile>
-            false
-            </useReleaseProfile>
-          </configuration>
-          <groupId>
-          org.apache.maven.plugins
-          </groupId>
-          <version>
-          2.5.3
-          </version>
-        </plugin>
-        <plugin>
-          <artifactId>
-          maven-source-plugin
-          </artifactId>
-          <configuration>
-            <skipSource>
-            true
-            </skipSource>
-          </configuration>
-          <groupId>
-          org.apache.maven.plugins
-          </groupId>
-          <version>
-          3.0.1
-          </version>
-        </plugin>
-        <plugin>
-          <artifactId>
-          maven-javadoc-plugin
-          </artifactId>
-          <configuration>
-            <skip>
-            true
-            </skip>
-          </configuration>
-          <groupId>
-          org.apache.maven.plugins
-          </groupId>
-          <version>
-          3.0.0
-          </version>
-        </plugin>
-      </plugins>
-    </pluginManagement>
-  </build>
   <dependencyManagement>
     <dependencies>
       <dependency>
@@ -880,7 +814,7 @@
       https://webtide.com
       </organizationUrl>
       <timezone>
-      -6
+      -7
       </timezone>
     </developer>
     <developer>
@@ -920,6 +854,26 @@
       -8
       </timezone>
     </developer>
+    <developer>
+      <email>
+      oliver.lamy@gmail.com
+      </email>
+      <id>
+      olamy
+      </id>
+      <name>
+      Olivier Lamy
+      </name>
+      <organization>
+      Webtide, LLC
+      </organization>
+      <organizationUrl>
+      https://webtide.com
+      </organizationUrl>
+      <timezone>
+      Australia/Brisbane
+      </timezone>
+    </developer>
   </developers>
   <distributionManagement>
     <repository>
@@ -938,7 +892,7 @@
       jetty.eclipse.website
       </id>
       <url>
-      scp://build.eclipse.org:/home/data/httpd/download.eclipse.org/jetty/${project.version}/
+      scp://build.eclipse.org:/home/data/httpd/download.eclipse.org/jetty/9.4.14-SNAPSHOT/jetty-bom/
       </url>
     </site>
     <snapshotRepository>
@@ -985,85 +939,90 @@
       </url>
     </license>
   </licenses>
+  <mailingLists>
+    <mailingList>
+      <archive>
+      https://dev.eclipse.org/mhonarc/lists/jetty-dev/maillist.html
+      </archive>
+      <name>
+      Jetty Developer Mailing List
+      </name>
+      <subscribe>
+      https://dev.eclipse.org/mailman/listinfo/jetty-dev
+      </subscribe>
+      <unsubscribe>
+      https://dev.eclipse.org/mailman/listinfo/jetty-dev
+      </unsubscribe>
+    </mailingList>
+    <mailingList>
+      <archive>
+      https://dev.eclipse.org/mhonarc/lists/jetty-commit/maillist.html
+      </archive>
+      <name>
+      Jetty Commit Mailing List
+      </name>
+      <subscribe>
+      https://dev.eclipse.org/mailman/listinfo/jetty-commit
+      </subscribe>
+      <unsubscribe>
+      https://dev.eclipse.org/mailman/listinfo/jetty-commit
+      </unsubscribe>
+    </mailingList>
+    <mailingList>
+      <archive>
+      https://dev.eclipse.org/mhonarc/lists/jetty-users/maillist.html
+      </archive>
+      <name>
+      Jetty Users Mailing List
+      </name>
+      <subscribe>
+      https://dev.eclipse.org/mailman/listinfo/jetty-users
+      </subscribe>
+      <unsubscribe>
+      https://dev.eclipse.org/mailman/listinfo/jetty-users
+      </unsubscribe>
+    </mailingList>
+    <mailingList>
+      <archive>
+      https://dev.eclipse.org/mhonarc/lists/jetty-announce/maillist.html
+      </archive>
+      <name>
+      Jetty Announce Mailing List
+      </name>
+      <subscribe>
+      https://dev.eclipse.org/mailman/listinfo/jetty-announce
+      </subscribe>
+      <unsubscribe>
+      https://dev.eclipse.org/mailman/listinfo/jetty-announce
+      </unsubscribe>
+    </mailingList>
+  </mailingLists>
   <modelVersion>
   4.0.0
   </modelVersion>
   <name>
   Jetty :: Bom
   </name>
+  <organization>
+    <name>
+    Webtide
+    </name>
+    <url>
+    https://webtide.com
+    </url>
+  </organization>
   <packaging>
   pom
   </packaging>
-  <profiles>
-    <profile>
-      <build>
-        <plugins>
-          <plugin>
-            <artifactId>
-            maven-deploy-plugin
-            </artifactId>
-            <configuration>
-              <updateReleaseInfo>
-              true
-              </updateReleaseInfo>
-            </configuration>
-            <groupId>
-            org.apache.maven.plugins
-            </groupId>
-            <inherited>
-            true
-            </inherited>
-          </plugin>
-          <plugin>
-            <artifactId>
-            maven-gpg-plugin
-            </artifactId>
-            <executions>
-              <execution>
-                <goals>
-                  <goal>
-                  sign
-                  </goal>
-                </goals>
-                <id>
-                sign-artifacts
-                </id>
-                <phase>
-                verify
-                </phase>
-              </execution>
-            </executions>
-            <groupId>
-            org.apache.maven.plugins
-            </groupId>
-          </plugin>
-        </plugins>
-      </build>
-      <id>
-      eclipse-release
-      </id>
-    </profile>
-  </profiles>
-  <properties>
-    <build-support.version>
-    1.4
-    </build-support.version>
-    <jetty.url>
-    http://www.eclipse.org/jetty
-    </jetty.url>
-    <project.build.sourceEncoding>
-    UTF-8
-    </project.build.sourceEncoding>
-  </properties>
   <scm>
     <connection>
-    scm:git:https://github.com/eclipse/jetty.project.git
+    scm:git:https://github.com/eclipse/jetty.project.git/jetty-bom
     </connection>
     <developerConnection>
-    scm:git:git@github.com:eclipse/jetty.project.git
+    scm:git:git@github.com:eclipse/jetty.project.git/jetty-bom
     </developerConnection>
     <url>
-    https://github.com/eclipse/jetty.project
+    https://github.com/eclipse/jetty.project/jetty-bom
     </url>
   </scm>
   <url>
```

### CODE REVIEW

**Comment by stevenschlansker** [comment]
There is an alternate version at https://github.com/eclipse/jetty.project/commit/88b97e9ae66e712c734b5b1c06071a8e1f4ab6c4 against master.

**Comment by joakime** [comment]
We are evaluating this approach.

File: jetty-bom/pom.xml [hunk 1]
```diff
@@ -1,92 +1,51 @@
...
-        </plugin>
-      </plugins>
-    </pluginManagement>
+    <plugins>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>flatten-maven-plugin</artifactId>
+        <version>1.0.1</version>
+        <configuration>
+          <outputDirectory>${project.build.director}</outputDirectory>
```
Comment (olamy) [inline jetty-bom/pom.xml]: typo here missing y must be `${project.build.directory}`

**Comment by olamy** [comment]
you need to rebase from jetty-9.4.x 
but very good fix!

**Comment by stevenschlansker** [comment]
@olamy thanks for catching the typo -- updated, rebased, and signed.  Take another look!

**Comment by olamy** [comment]
@stevenschlansker thanks for your contribution!

**Comment by joakime** [comment]
@stevenschlansker thanks for the patch.  Want one in return?
https://webtide.com/patch-for-a-patch/

**Pull request merged**